python/triton/language/core.py

```diff
@@ -884,55 +884,87 @@ def dot(input, other, allow_tf32=True, out_dtype=float32, _builder=None):
 
 
 @builtin
-def load(pointer, mask=None, other=None, cache_modifier="", eviction_policy="", volatile=False, _builder=None):
+def load(pointer, mask=None, other=None, boundary_check=(), padding_option="", cache_modifier="", eviction_policy="",
```

Comment:
I'm not convinced the load and store function need to be so complicated. You could have two functions `_unstructured_load` (exactly the old code path) and `_structured_load` (new code path). Peak the right code path depending on the type of `pointer`. `tensor<ptr> -> _unstructured` and `ptr<tensor> -> structured`